src/main/java/seedu/address/commons/util/CsvUtil.java

```diff
@@ -0,0 +1,123 @@
+        for (int i = 0; i < headers.length; i++) {
+            headerIndex.put(headers[i], i);
+        }
+
+        ArrayList<T> list = new ArrayList<T>();
+        for (int i = 1; i < csvData.size(); i++) {
+            String[] fields = csvData.get(i);
+            T temp;
+            try {
+                String name = fields[headerIndex.get("Name")];
+                String tele = StringUtil.clean(fields[headerIndex.get("Telegram")], "@");
+                String github = StringUtil.clean(fields[headerIndex.get("Github")], "@");
+                String phone = fields[headerIndex.get("Phone Number")];
+                String email = fields[headerIndex.get("Email")];
+                String address = StringUtil.clean(fields[headerIndex.get("Address")], "\"");
+                List<String> tags = Arrays.asList(fields[headerIndex.get("Tags")].split(" "));
+                temp = constructorOfObjectToDeserialize.newInstance(name, tele, github, phone, email, address, tags);
+            } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+                temp = null;
+            }
+            list.add(temp);
+        }
+        return list;
+    }
+
+    /**
+     * Converts Data in CSV file to a JSON String
+     *
+     * @param constructorOfObjectToDeserialize The constructor for T object
+     * @param filePath path of CSV file
+     * @param <T> The generic type to create an instance of
+     * @return a JSON String
+     * @throws IOException if there was an error reading file
+     */
+    public static <T> String csvToJsonString(Constructor<T> constructorOfObjectToDeserialize, Path filePath)
+            throws IOException {
+        PersonWrapper<T> data = new PersonWrapper<>(csvToPersonData(constructorOfObjectToDeserialize, filePath));
+        return JsonUtil.toJsonString(data);
+    }
+
+    /***
+     * Reads data in CSV file
+     *
+     * @param filePath path of CSV file to read
+     * @return an ArrayList of String Array containing data split by comma
+     * @throws IOException if there was an error reading file
+     */
+    public static ArrayList<String[]> readCsv(Path filePath) throws IOException {
+        String file = FileUtil.readFromFile(filePath);
+        String regex = ",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)";
+        return Arrays.stream(file.split("\n"))
+                .map(s1 -> Arrays.stream(s1.split(regex)).sequential()
+                        .map(StringUtil::clean)
+                        .collect(Collectors.toList()).toArray(new String [0]))
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * Converts a CSV File to a JSON File
+     * @param constructorOfObjectToDeserialize The constructor for T object
+     * @param csvFilePath path of CSV file
+     * @param jsonFilePath path of JSON file
+     * @param <T> The generic type to create an instance of
+     * @throws IOException if there was an error reading file
+     */
```

Comment:
Slight difference in the indentation of Javadoc from the previous method, missing one empty line between lines 102 and 103. It would be better to follow the same format throughout 😃